### PR TITLE
skip active connections check when downstream cx stats are not available

### PR DESCRIPTION
**Pull request by ramaraochavali**
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [X] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/envoy/agent.go [hunk 1]
```diff
@@ -149,6 +149,11 @@ func (a *Agent) terminate() {
 		ticker := time.NewTicker(activeConnectionCheckDelay)
 		for range ticker.C {
 			ac := a.activeProxyConnections()
```
Comment (howardjohn) [inline pkg/envoy/agent.go:151]: did you check it will still show up when zero? since usedonly was set I wasn't 100% sure

Comment (ramaraochavali) [reply pkg/envoy/agent.go:151]: Yes.  That is correct. But my thought process was if downstream_cx_active is not there at all in the stats, we can skip active connections. May be we should reword the error message?

File: pkg/envoy/agent.go [hunk 2]
```diff
@@ -149,6 +149,12 @@ func (a *Agent) terminate() {
...
+			if ac == -1 {
+				log.Info("downstream_cx_active are not available. This either means there are no downstream connection established yet" +
+					" or the stats are not enabled. Skipping active connections check...")
```
Comment (hzxuzhonghu) [inline pkg/envoy/agent.go:154]: In theory it could be an error too

Comment (ramaraochavali) [reply pkg/envoy/agent.go:154]: I did not change it to error because no client might have connected to it.

Comment (hzxuzhonghu) [reply pkg/envoy/agent.go:154]: I mean it could be an error calling 127.0.0.1:15000

Comment (ramaraochavali) [reply pkg/envoy/agent.go:159]: I see. Handled error differently

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**